### ISSUE TITLE
Worker could not be restarted until all keep-alive connections are closed

### ISSUE THREAD

**Issue by narqo** [nodules/luster#36]
`WorkerWrapper` uses [`worker.disconnect()`](https://nodejs.org/dist/latest-v4.x/docs/api/cluster.html#cluster_worker_disconnect) internally, which waits a server to be closed before `disconnect` event would be fired. When the server accepts persistent connections (e.g. keep-alive) it doesn't fire `close` until the last connection is closed (see nodejs/node#2642 for discussion).

This leads to a situation when worker is hanging in the `restarting` state for minutes (by default keepalive connection lives ~2mins). During this time worker isn't able to process new connections.

``` js
// worker.js

http.createServer((req, res) => res.end('ok')).listen(process.env.port);
```

```
› env LUSTER_CONF='app=./worker.js; workers=1; server.port=8000' luster ... &
› curl --keepalive http://localhost:8000
```

If master process calls `restart`, worker will hang.

There are several possible solutions that I can think of:
1. luster has `control.stopTimeout` option with which worker is force killed after a timeout during the stopping process. Unfortunately it only works **after** `disconnect` event was emitted in worker, which isn't in the described case. We can change the logic behind `stopTimeout`, so it to start the timer right after `worker.disconnect()` was called. This, actually the behaviour that is described in current Node.js docs (see example below [cluster_worker_disconnect](https://nodejs.org/dist/latest-v4.x/docs/api/cluster.html#cluster_worker_disconnect)).
2. luster can fire some public event (`SIGHUP` maybe) to signal the process is going to be restarted. With this approach the worker could use it's own custom logic to do a graceful restarting. For example server could use [http-shutdown][2] module to call `server.shutdown()` which closes all keep alive connections that are not active.

**Comment by narqo** [comment]
Possible related to nodules/luster-guard#11

**Comment by kaero** [comment]
I prefer the first option.

**Comment by SlNPacifist** [comment]
I'll check this soon. Most likely, this was fixed in 1.0.0 as `stopTimeout` does not wait for `disconnect` event anymore.

**Comment by SlNPacifist** [comment]
Fixed in 1.0.0: `stopTimeout` does not wait for `disconnect` event.
